src/main/java/randoop/condition/SpecificationTranslator.java

```diff
@@ -0,0 +1,388 @@
+
+  /** The name of dummy variables used by {@link randoop.contract.ObjectContract}. */
+  private static final String DUMMY_VARIABLE_NAME = "x";
+
+  /** The {@link RawSignature} for a guard expression method. */
+  private RawSignature guardExpressionSignature;
+
+  /** The parameter declaration string, with parentheses, for a guard expression method. */
+  private final String guardExpressionDeclaration;
+
+  /** The {@link RawSignature} for a property expression method. */
+  private RawSignature propertyExpressionSignature;
+
+  /** The parameter declaration string, with parentheses, for a property expression method. */
+  private final String propertyExpressionDeclarations;
+
+  /** The map of expression identifiers to dummy variables. */
+  private final NameReplacementMap replacementMap;
+
+  /** The {@link SequenceCompiler} for compiling expression methods. */
+  private final SequenceCompiler compiler;
+
+  /**
+   * Creates a {@link SpecificationTranslator} object in the given package with the signature
+   * strings and variable replacementMap.
+   *
+   * @param guardExpressionSignature the {@link RawSignature} for a guard expression method
+   * @param guardExpressionDeclaration the parameter declaration string, with parentheses, for a
+   *     guard expression method
+   * @param propertyExpressionSignature the {@link RawSignature} for a property expression method
+   * @param propertyExpressionDeclaration the parameter declaration string, with parentheses, for a
+   *     property expression method
+   * @param replacementMap the map of expression identifiers to dummy variables
+   * @param compiler the {@link SequenceCompiler} for creating expression methods
+   */
+  private SpecificationTranslator(
+      RawSignature guardExpressionSignature,
+      String guardExpressionDeclaration,
+      RawSignature propertyExpressionSignature,
+      String propertyExpressionDeclaration,
+      NameReplacementMap replacementMap,
+      SequenceCompiler compiler) {
+    this.guardExpressionSignature = guardExpressionSignature;
+    this.guardExpressionDeclaration = guardExpressionDeclaration;
+    this.propertyExpressionSignature = propertyExpressionSignature;
+    this.propertyExpressionDeclarations = propertyExpressionDeclaration;
+    this.replacementMap = replacementMap;
+    this.compiler = compiler;
+  }
+
+  /**
+   * Creates a {@link SpecificationTranslator} object to translate the {@link
+   * OperationSpecification} of {@code executable}.
+   *
+   * @param executable the {@code java.lang.reflect.AccessibleObject} for the operation with {@link
+   *     OperationSpecification} to translate
+   * @param identifiers the {@link Identifiers} from the specification to be translated
+   * @param compiler the sequence compiler to use to create expression methods
+   * @return the translator object to convert the specifications for {@code executable}
+   */
+  static SpecificationTranslator createTranslator(
```

Comment:
Why isn't this a constructor?
Factory methods are usually used to create a value of a different type, to return a possibly-existing value, or to disambiguate if there would be too many constructors with confusingly-similar signatures.  What is the reason this isn't a factory method?